crates/rune/src/runtime/any_obj.rs

```diff
@@ -82,6 +83,53 @@ impl AnyObj {
         }
     }
 
+    /// Create an AnyObj from a reference to a type T containing lifetimes.
+    pub(crate) fn new_projection<T>(value: &T) -> Self
+    where
+        T: AnyRef,
+    {
+        let projection = Projection::new(value);
+
+        let data =
+            unsafe { ptr::NonNull::new_unchecked(Box::into_raw(Box::new(projection)) as *mut ()) };
```

Comment:
The `Projection` is itself a `ptr::NonNull`, which I think is unnecessary. It can probably stored as the `T: AnyRef` itself, along with a new `AnyObjKind`. And then the `Projection` only needs to be returned when accessing the object.